src/dataflow-types/src/types.rs

```diff
@@ -1361,10 +1394,156 @@ pub mod sources {
         }
 
         pub fn options(&self) -> BTreeMap<String, String> {
+            const SECURITY_PROTOCOL: &str = "security.protocol";
             match self {
-                ConnectorInner::Kafka { config_options, .. } => config_options.clone(),
-                ConnectorInner::CSR { with_options, .. } => with_options.clone(),
+                ConnectorInner::CSR {
+                    username, password, ..
+                } => {
+                    let mut with_options = BTreeMap::new();
+                    if let Some(username) = username {
+                        with_options.insert("username".to_string(), username.to_owned());
+                    }
+                    if let Some(password) = password {
+                        with_options.insert("password".to_string(), password.to_owned());
+                    }
+                    with_options
+                }
+                // Initially we are just going to convert the new struct back into the old map so downstream code can stay unchanged
+                ConnectorInner::Kafka { security, .. } => match security {
+                    KafkaSecurityOptions::PLAINTEXT => {
+                        let mut with_options = BTreeMap::new();
+                        with_options.insert(SECURITY_PROTOCOL.to_string(), "PLAINTEXT".to_string());
+                        with_options
+                    }
+                    KafkaSecurityOptions::SSL {
+                        key,
+                        certificate,
+                        passphrase,
+                    } => {
+                        let mut with_options = BTreeMap::new();
+                        with_options.insert(SECURITY_PROTOCOL.to_string(), "SSL".to_string());
+                        if let Some(uuid) = key {
+                            with_options.insert(
+                                "ssl.key.location".to_string(),
+                                format!("secrets/{}", uuid),
```

Comment:
Agreed, this absolutely needs to not be hardcoded. In my local branch I actually added a second `canonical_path(uuid: GlobalId) -> PathBuf` method to the SecretsReader trait so that all of this could be encapsulated in the secrets code.